contribs/gmf/src/directives/timeslider.js

```diff
@@ -0,0 +1,182 @@
+goog.provide('gmf.TimeSliderDirective');
+goog.provide('gmf.TimeSliderController');
+
+goog.require('gmf');
+
+gmf.module.value('gmfTimeSliderTemplateUrl',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     * @return {string} Template URL.
+     */
+    function(element, attrs) {
+      var templateUrl = attrs['gmfTimeSliderTemplateUrl'];
+      return templateUrl !== undefined ? templateUrl :
+          gmf.baseTemplateUrl + '/timeslider.html';
+    });
+
+
+/**
+ * Provide a directive to select a signle date or a range of dates with a slider
+ * @param {string|function(!angular.JQLite=, !angular.Attributes=)}
+ * gmfTimeSliderTemplateUrl Template for the directive.
+ * @param {angular.$timeout} $timeout angular timeout service
+ * @param {angular.$filter} $filter angular filter service
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfTimeSlider
+ */
+gmf.TimeSlider = function(gmfTimeSliderTemplateUrl, $timeout, $filter) {
+  return {
+    scope : {
+      onDateSelected : '&',
+      time : '='
+    },
+    bindToController : true,
+    controller : 'gmfTimeSliderController',
+    controllerAs : 'sliderCtrl',
+    restrict: 'AE',
+    templateUrl : gmfTimeSliderTemplateUrl,
+    link: function(scope, element, attrs, ctrl) {
+
+      /** @type {string} */
+      var dateFormat = 'shortDate';
+
+      ctrl.sliderOptions['slide'] = updateTooltip_;
+      ctrl.sliderOptions['start'] = updateTooltip_;
+      ctrl.sliderOptions['stop'] = onSliderReleased_;
+      $timeout(init_);
```

Comment:
because of https://github.com/angular-ui/ui-slider/blob/master/src/slider.js#L175 
...